
Removed code prevented "Satisfy Any" directive from working, cupsdCheckPolicy
checks for auth info AND allow/deny access lists and does its job properly
so checking again just for auth info is The Wrong Thing as there may be
an access list allowing unauthorized clients based on their IP.

--- cups-1.3.8/scheduler/ipp.c~	2008-07-23 14:09:50.444066834 +0200
+++ cups-1.3.8/scheduler/ipp.c	2008-07-23 14:11:36.356848959 +0200
@@ -1301,13 +1301,7 @@
     send_http_error(con, status, printer);
     return (NULL);
   }
-  else if (printer->num_auth_info_required > 0 &&
-           strcmp(printer->auth_info_required[0], "none") &&
-           !con->username[0] && !auth_info)
-  {
-    send_http_error(con, HTTP_UNAUTHORIZED, printer);
-    return (NULL);
-  }
+
 #ifdef HAVE_SSL
   else if (auth_info && !con->http.tls &&
            !httpAddrLocalhost(con->http.hostaddr))
